src/Valleysoft.DockerCredsProvider.Test/CredsProviderTests.cs

```diff
@@ -490,12 +433,155 @@ public async Task ConfigFileDoesNotExist()
+        string fullHelperName = $"docker-credential-{helper}";
+        string username = "<token>";
+        string token = "token";
+
+        string dockerConfigContent =
+            "{" +
+                "\"credHelpers\": {" +
+                    $"\"testregistry\": \"{helper}\"" +
+                "}" +
+            "}";
+
+        // the idea here is that we setup the helper on the second PATH entry,
+        // so if we succeed that means we probed.
+        var systemPaths = new List<string>{
+            "/a",
+            "/b"
+        };
+
+        Mock<IFileSystem> fileSystemMock = new();
+        fileSystemMock
+            .WithFile(dockerConfigPath, dockerConfigContent)
+            .WithFile(Path.Combine(systemPaths[1], fullHelperName));
+
+        Mock<IEnvironment> envMock = new();
+        envMock.WithSystemProfileFolder();
+        envMock.WithPath(systemPaths);
+        envMock.Setup(o => o.GetEnvironmentVariable("PATHEXT")).Returns<string>(null);
+
+        Mock<IProcessService> processServiceMock = new();
+        processServiceMock.StubHelperSuccess(helper, "testregistry",  $"{{ \"Username\": \"{username}\", \"Secret\": \"{token}\" }}");
+
+        DockerCredentials creds = await CredsProvider.GetCredentialsAsync("testregistry", fileSystemMock.Object, processServiceMock.Object, envMock.Object);
+
+        Assert.Equal(username, creds.Username);
+        Assert.Equal(token, creds.IdentityToken);
+        Assert.Null(creds.Password);
+    }
+
+    [WindowsFact]
+    public async Task NativeStore_ProbesPATHEXTForHelperBinaries() {
+        string dockerConfigPath = Path.Combine(
+            Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
+            ".docker",
+            "config.json");
+
+        string helper = "example";
+        string fullHelperName = $"docker-credential-{helper}";
+        string username = "<token>";
+        string token = "token";
+
+        string dockerConfigContent =
+            "{" +
+                "\"credHelpers\": {" +
+                    $"\"testregistry\": \"{helper}\"" +
+                "}" +
+            "}";
+
+        var pathRoot = "/a";
+        // the idea here is that we set up the helper with a different extension
+        // and prime the system to probe that extension.  if we succeed, that means we
+        // probed as expected.
+        var pathExts = new List<string>{
+            ".ABC",
+            ".XYZ"
+        };
+
+        Mock<IFileSystem> fileSystemMock = new();
+        fileSystemMock
+            .WithFile(dockerConfigPath, dockerConfigContent)
+            .WithFile(Path.Combine(pathRoot, $"{fullHelperName}{pathExts[0]}"))
+            .WithFile(Path.Combine(pathRoot, $"{fullHelperName}{pathExts[1]}"));
+
+        Mock<IEnvironment> envMock = new();
+        envMock.WithSystemProfileFolder();
+        envMock.WithPath(new List<string> { pathRoot });
+        envMock.WithExecutableExtensions(pathExts);
```

Comment:
```suggestion
        envMock.WithSystemProfileFolder()
            .WithPath(new List<string> { pathRoot })
            .WithExecutableExtensions(pathExts);
```

Reply:
Applied this in a couple other places I saw that I'd missed as well.